libtenzir/builtins/functions/misc.cpp

```diff
@@ -48,25 +51,134 @@ class secret final : public function_plugin {
     return "tql2.secret";
   }
 
+  auto initialize(const record& plugin_config, const record& global_config)
+    -> caf::error override {
+    TENZIR_UNUSED(plugin_config);
+    auto secrets = try_get_or(global_config, "tenzir.secrets", record{});
+    if (not secrets) {
+      return diagnostic::error(secrets.error())
+        .note("configuration key `tenzir.secrets` must be a record")
+        .to_error();
+    }
+    for (const auto& [key, value] : *secrets) {
+      const auto* str = caf::get_if<std::string>(&value);
+      if (not str) {
+        return diagnostic::error("secrets must be strings")
+          .note("configuration key `tenzir.secrets.{}` is of type `{}`", key,
+                type::infer(value).value_or(type{}).kind())
+          .to_error();
+      }
+      secrets_.emplace(key, *str);
+    }
+    return {};
+  }
+
   auto make_function(invocation inv, session ctx) const
     -> failure_or<function_ptr> override {
     auto expr = ast::expression{};
-    TRY(argument_parser2::function("secret")
-          .add(expr, "<string>")
-          .parse(inv, ctx));
-    return function_use::make(
-      [expr = std::move(expr)](evaluator eval, session ctx) -> series {
-        TENZIR_UNUSED(ctx);
-        auto value = eval(expr);
-        auto b = arrow::StringBuilder{};
-        check(b.Reserve(value.length()));
-        for (auto i = int64_t{0}; i < value.length(); ++i) {
-          // TODO: Actually resolve the secret.
-          check(b.Append("A385F-BB052-CAFE9-CEC2D"));
-        }
-        return {string_type{}, finish(b)};
-      });
+    TRY(
+      argument_parser2::function("secret").add(expr, "<key>").parse(inv, ctx));
```

Comment:
We could require a constant for the secret name here (same for `env`).

Reply:
Can we transparently support both?

Reply:
Would need some small changes to the evaluator.

Reply:
I wrote a follow-up issue for this, resolving this for now so that I can merge this.